cmake/macros/macro_deal_ii_setup_target.cmake

```diff
@@ -53,13 +53,32 @@ macro(deal_ii_setup_target _target)
   endif()
 
   #
-  # Set build type with the help of the specified keyword, or
-  # CMAKE_BUILD_TYPE:
+  # Parse the optional arguments passed to this macro. These
+  # may only contain the build type (DEBUG or RELEASE).
   #
+  set(_argn_build_type)
+  foreach (_arg ${ARGN})
+    if("${_arg}" MATCHES "^(DEBUG|RELEASE)$")
+      set(_argn_build_type ${_arg})
+    else()
+      message(FATAL_ERROR
+        "\nThe deal_ii_setup_target() macro was called with an invalid argument. "
+        "Valid arguments are (none), DEBUG, or RELEASE. "
+        "The argument given is <${_arg}>."
+        "\n\n"
+        )
+    endif()
+  endforeach()
 
-  if("${ARGN}" MATCHES "^(DEBUG|RELEASE)$")
-    set(_build "${ARGN}")
-  elseif("${ARGN}" STREQUAL "")
+  #
+  # Set build type. It can either be specified explicitly as an
+  # additional macro argument that must be either DEBUG or RELEASE
+  # (parsed above); alternatively, failing a specific optional
+  # argument, it is set based on CMAKE_BUILD_TYPE:
+  #
+  if(_argn_build_type)
+    set(_build "${_argn_build_type}")
+  else()
```

Comment:
```suggestion
  if("${_build}" STREQUAL "")
```

Reply:
No matter what you decide to do, please don't use `if(_argn_build_type)` directly.

CMake defines the following behavior: "True if given a variable that is defined to a value that is not a false constant." I understand that this does the right thing in this context. But it does so by conflating booleans and strings... and it will break the moment someone has the glorious idea to define a build type "False".

So the right thing to do is to simply check whether `"${_argn_build_type}"` expands to a nonzero string. That is the reason why you see so many `STREQUAL` checks in the cmake code...

Reply:
OK, yes, you know how I feel about being explicit and using strong types :-)